SwrveSDKTest/src/testGoogle/java/com/swrve/sdk/SwrveGoogleUnitTest.java

```diff
@@ -0,0 +1,128 @@
+package com.swrve.sdk;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+
+import com.swrve.sdk.config.SwrveConfig;
+import com.swrve.sdk.gcm.SwrveGcmConstants;
+import com.swrve.sdk.test.BuildConfig;
+import com.swrve.sdk.test.MainActivity;
+import com.swrve.sdk.utils.TestHttpServer;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.Shadows;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowActivity;
+import org.robolectric.shadows.ShadowLog;
+
+import java.lang.reflect.Field;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Iterator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(RobolectricGradleTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+@TargetApi(Build.VERSION_CODES.GINGERBREAD)
+public class SwrveGoogleUnitTest extends SwrveBaseTest {
+
+    private Activity activity;
+
+    @Before
+    public void setUp() throws Exception {
+        removeSingletonInstance();
+        activity = Robolectric.buildActivity(MainActivity.class).create().visible().get();
+        ShadowLog.stream = System.out;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        removeSingletonInstance();
+    }
+
+    private void removeSingletonInstance() throws Exception {
```

Comment:
I've just merged in this same method here:
https://github.com/Swrve/swrve-android-sdk/blob/release-4_6/SwrveSDKTest/src/test/java/com/swrve/sdk/SwrveTestUtils.java#L39